docs/src/validator/best-practices/operations.md

```diff
@@ -0,0 +1,133 @@
+
+The Solana validator community holds regular educational workshops. You can watch past workshops through the [solana validator educational workshops playlist](https://www.youtube.com/watch?v=86zySQ5vGW8&list=PLilwLeBwGuK6jKrmn7KOkxRxS9tvbRa5p).
+
+## Help with the validator command line
+
+From within the Solana CLI, you can execute the `solana-validator` command with the `--help` flag to get a better understanding of the flags and sub commands available.
+
+```
+solana-validator --help
+```
+
+## Restarting your validator
+
+There are many operational reasons you may want to restart your validator. As a best practice, you should avoid a restart during a leader slot. A [leader slot](../../terminology.md#leader-schedule) is the time when your validator is expected to produce blocks. For the health of the cluster and also for your validator's ability to earn transaction fee rewards, you do not want your validator to be offline during an opportunity to produce blocks.
+
+To see the full leader schedule for an epoch, use the following command:
+
+```
+solana leader-schedule
+```
+
+Based on the current slot and the leader schedule, you and calculate open time windows where your validator is not expected to produce blocks.
+
+Assuming you are ready to restart, you may use the `solana-validator exit` command.  The command exits your validator process when an appropriate idle time window is reached.  Assuming that you have systemd implemented for your validator process, the validator should restart automatically after the exit.  See the below help command for details:
+
+```
+solana-validator exit --help
+```
+
+## Upgrading
+
+There are many ways to upgrade the [Solana software](../../cli/install-solana-cli-tools.md). As an operator, you will need to upgrade often, so it is important to get comfortable with this process.
+
+> **Note** validator nodes do not need to be offline while the newest version is being downloaded or built from source.  All methods below can be done before the validator process is restarted.
+
+### Building From Source
+
+It is a best practice to always build your Solana binaries from source. If you build from source, you are certain that the code you are building has not been tampered with before the binary was created. You may also be able to optimize your `solana-validator` binary to your specific hardware.
+
+If you build from source on the validator machine (or a machine with the same CPU), you can target your specific architecture using the `-march` flag. Refer to the Solana docs for [instructions on building from source](../../cli/install-solana-cli-tools.md#build-from-source).
+
+### solana-install
+
+If you are not comfortable building from source, or you need to quickly install a new version to test something out, you could instead try using the `solana-install` command.
+
+Assuming you want to install Solana version `1.14.17`, you would execute the following:
+
+```
+solana-install init 1.14.17
```

Comment:
this wont work on the first install so perhaps worth adding the curl option in

Reply:
I added a note below.  Let me know if you think that works.  This section is mainly about operations so ideally they have at least installed the cli already.

Reply:
Yeah that makes sense.

Reply:
> I added a note below. Let me know if you think that works. This section is mainly about operations so ideally they have at least installed the cli already.

Yeah, that makes sense.